.github/actions/archive-charm-testing-artifacts/action.yaml

```diff
@@ -0,0 +1,33 @@
+name: archive-charm-testing-artifacts
+description: Archives the charm(s) that were built during a job, along with possible log and juju crashdump outputs
+inputs:
+  enable_containerd_template_adjustment:
+    description: 'Whether to adjust containerd template for GHCR.io credentials'
+    required: false
+  ghcr_package_api_user:
+    description: 'GitHub Container Registry API User'
+    required: false
+  ghcr_package_api_token:
+    description: 'GitHub Container Registry API Token'
+    required: false
+    
+runs:
+  using: 'composite'
+  steps:
+  - name: Archive the tested charm
+    uses: actions/upload-artifact@v3
+    with:
+      name: tested-charm
```

Comment:
given that we pass the name here, does that mean if we run this action twice, it will overwrite the previous archived artifacts? e.g. if both frontend and api of test observer build would only one show artefacts?

Reply:
I now added a `github.run_id` into the name now as well to make them unique by the workflow run. I don't think there would be overwriting anyway but makes sense to do either way.